### PR TITLE
Move UrbanAirship to Eliza

### PR DESCRIPTION
**Pull request by raymondng**
registerDevice is merely commented out -- please let me know if it's used anywhere.

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala [hunk 1]
```diff
@@ -306,6 +308,10 @@ class MessagingController @Inject() (
     new String(outBuf.array, 0, outBuf.position(), charset)
   }
 
+  def sendPushNotification(userId:Id[User], extId:String, pendingNotificationCount:Int, msg:String) = {
+    urbanAirship.notifyUser(userId, PushNotification(ExternalId[UserNotification](extId), pendingNotificationCount, msg))
```
Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala]: Can we change `ExternalId[UserNotification]` to `ExternalId[Message]`? We used the former for backwards compatibility in shoebox, but thats no longer needed.

**Comment by andrewconner** [comment]
It looks like register device is still needed. External calls come in to `/devices/` to register new devices. It likely needs to be on Eliza as well, but @ebfaed is the one to ask about that. We use this, right?

**Comment by raymondng** [comment]
Thanks. Funny thing is I did look into it -- I couldn't find it being used anywhere. If it's used then it could be in some meta-language or generated code.

**Comment by stkem** [comment]
I think you missed the FakeShoeboxServiceClient and the Database Evolutions.

**Comment by raymondng** [comment]
Thanks. Will look into it

**Comment by raymondng** [comment]
Thanks guys. Will follow up with greg about registerDevice usage.

**Comment by raymondng** [comment]
Thanks greg for solving the mystery. It _is_ being used by ios client. I probably asked Eduardo the wrong question when I asked him about "registerDevice" instead of "/devices" :(

I'll expose the endpoint via eliza and work with Eduardo on the ios-related changes.

**Comment by eishay** [comment]
nice!

**Comment by ebfio** [comment]
All iOS changes done!

Great work, Ray! 

-- 
Eduardo Fonseca

On September 6, 2013 at 2:44:14 PM, Raymond Ng (notifications@github.com) wrote:

Thanks greg for solving the mystery. It is being used by ios client. I probably asked Eduardo the wrong question when I asked him about "registerDevice" instead of "/devices" :(

I'll expose the endpoint via eliza and work with Eduardo on the ios-related changes.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala]: Thanks. I'll look into it. Probably in a follow-up request.

Comment (raymondng) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala]: It's been taken care of.

**Pull request merged**